source/features/toggle-everything-with-alt.tsx

```diff
@@ -1,32 +1,56 @@
 import mem from 'mem';
+import React from 'dom-chef';
 import select from 'select-dom';
 import delegate from 'delegate-it';
+import {CheckIcon} from '@primer/octicons-react';
 import * as pageDetect from 'github-url-detection';
 
 import features from '.';
 import preserveScroll from '../helpers/preserve-scroll';
+import progressNotification from '../github-helpers/progress-notification';
 
 type EventHandler = (event: delegate.Event<MouseEvent, HTMLElement>) => void;
-
-const clickAll = mem((selectorGetter: ((clickedItem: HTMLElement) => string)): EventHandler => {
+// eslint-disable-next-line import/prefer-default-export
+export const clickAll = mem((selectorGetter: ((clickedItem: HTMLElement) => string), displayProgress?: boolean): EventHandler => {
 	return event => {
 		if (event.altKey && event.isTrusted) {
 			const clickedItem = event.delegateTarget;
 
 			// `parentElement` is the anchor because `clickedItem` might be hidden/replaced after the click
 			const resetScroll = preserveScroll(clickedItem.parentElement!);
-			clickAllExcept(selectorGetter(clickedItem), clickedItem);
+			clickAllExcept(selectorGetter(clickedItem), clickedItem, displayProgress);
```

Comment:
But the progress is unrelated to this operation. I suggest 2 things:

- Have a self-contained toaster/spinner component, like: `const toast = new Toast()` and `toast.done()` to be called later, something like that. This is something that is likely to be reused in other batch features as well.
- Have the event handler in `batch-mark-files-as-viewed` call `new Toast(); clickAllBlahBlah; toast.done();`

Reply:
> * Have the event handler in [`batch-mark-files-as-viewed`](/sindresorhus/refined-github/blob/main/source/features/batch-mark-files-as-viewed.tsx) call `new Toast(); clickAllBlahBlah; toast.done();`

Going to give this only one attempt. (_all help wanted_) 🚑 

How do I make an event handler pass back to the main function (ClickedItem calls clickExcept) and then have it available to pass back? Do we have any feature examples I can look at?

Reply:
I think it looks something like:

```ts
const markAsViewedSelector = clickAll((): string => '.js-reviewed-checkbox', true)

function handlerblah () {
	showToast
	markAsViewedSelector()
	hideToast
}

Reply:
🤷 

I tried it but it does not want to mark them all if I do that

Reply:
The current code is not acceptable though. Most clickAll operations take milliseconds to run and don’t need the toast. The toast does not belong in clickAll.

Reply:
Done

Reply:
https://github.com/sindresorhus/refined-github/pull/4044/files takes almost 3 seconds and freezes that page 😞

Reply:
> [#4044 (files)](https://github.com/sindresorhus/refined-github/pull/4044/files) takes almost 3 seconds and freezes that page 😞

That's why the the toast is needed for toggle-all

Reply:
I tried I dont know. Currently the feture does not work and I cant figure out WHY?!